--- a/src/Makefile
+++ b/src/Makefile
@@ -3,6 +3,13 @@
 PKG_CONFIG ?= pkg-config
 SDL2_CFLAGS := $(shell $(PKG_CONFIG) --cflags sdl2)
 SDL2_LIBS := $(shell $(PKG_CONFIG) --libs sdl2)
+
+ifneq (,$(findstring w64-mingw32,$(CXX)))
+PLATFORM := MINGW CROSS
+ifneq (,$(findstring x86_64,$(CXX)))
+PLATFORM += 64
+endif
+endif
 
 PLATFORM= $(shell uname -s | tr '[:lower:]' '[:upper:]')
 PLATFORM_ARCH= $(shell uname -m)
@@ -47,7 +54,7 @@
 endif
 CLIENT_INCLUDES= $(INCLUDES) $(SDL2_CFLAGS) -Iinclude
 STD_LIBS= -static-libgcc -static-libstdc++
-CLIENT_LIBS= -mwindows $(STD_LIBS) -L$(WINBIN) -L$(WINLIB) -lSDL2 -lSDL2_image -lSDL2_mixer -lzlib1 -lopengl32 -lenet -lws2_32 -lwinmm
+CLIENT_LIBS= $(STD_LIBS) -L$(WINBIN) -L$(WINLIB) $(SDL2_LIBS) -lSDL2_image -lSDL2_mixer -lz -lopengl32 -lenet -lws2_32 -lwinmm
 else	
 ifneq (,$(findstring DARWIN,$(PLATFORM)))
 OSXMIN= 10.9
@@ -130,8 +137,8 @@
 SERVER_INCLUDES= -DSTANDALONE -Istandalone/shared -Istandalone/engine -Istandalone/game $(INCLUDES)
 ifneq (,$(findstring MINGW,$(PLATFORM)))
 SERVER_INCLUDES+= -Iinclude
-SERVER_LIBS= -mwindows $(STD_LIBS) -L$(WINBIN) -L$(WINLIB) -lzlib1 -lenet -lws2_32 -lwinmm
-MASTER_LIBS= $(STD_LIBS) -L$(WINBIN) -L$(WINLIB) -lzlib1 -lenet -lws2_32 -lwinmm
+SERVER_LIBS= -mwindows $(STD_LIBS) -L$(WINBIN) -L$(WINLIB) -lz -lenet -lws2_32 -lwinmm
+MASTER_LIBS= $(STD_LIBS) -L$(WINBIN) -L$(WINLIB) -lz -lenet -lws2_32 -lwinmm
 else
 SERVER_LIBS= -lenet -lz
 MASTER_LIBS= $(SERVER_LIBS)
